Sage/carbon
# test: dialog

### Proposed behaviour

Refactor for Dialog and Dialog-Full-Screen components to use new cypress-component-react framework for testing.

### Current behaviour

Currently we are using the regular cypress approach with BDD.
### Checklist




#### QA


### Testing instructions

- Run npx cypress open-ct to check if there is newly added test.file
 - Check if the dialog.test.js file passed
 - Check if the dialog-full-screen.test.js file passed
 - Compare all tests were moved/refactored from .feature file -> .test.js file
 - Check if all unnecessary files were deleted, including .json file
